manifests/istio-control/istio-discovery/templates/configmap.yaml

```diff
@@ -256,14 +253,22 @@ data:
     {{- if not (eq .Values.revision "") }}
     {{- $defPilotHostname := printf "istiod-%s.%s.svc" .Values.revision .Release.Namespace }}
     {{- else }}
-    {{- $defPilotHostname := printf "istiod.%s.svc" .Release.Namespace }}
+    {{- $defPilotHostname := printf "istiod.%s.svc"  .Release.Namespace }}
     {{- end }}
     {{- $defPilotHostname := printf "istiod%s.%s.svc" .Values.revision .Release.Namespace }}
```

Comment:
defining variables like this doesn't actually work inside if statements. this is fixed on 1.5 but somehow broken on master, I have a fix in https://github.com/istio/istio/pull/21788/files

Reply:
Yup. Looks like your PR and my change below are similar except for the user of the DNS name instead of IP address.

Reply:
interesting, thanks!